nav2_util/src/CMakeLists.txt

```diff
@@ -10,34 +10,41 @@ add_library(${library_name} SHARED
   odometry_utils.cpp
   array_parser.cpp
 )
-
-ament_target_dependencies(${library_name}
-  rclcpp
-  nav2_msgs
-  tf2
-  tf2_ros
-  nav_msgs
-  geometry_msgs
-  lifecycle_msgs
-  rclcpp_lifecycle
-  tf2_geometry_msgs
-  bondcpp
+target_include_directories(${library_name}
+  PUBLIC
+  "$<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>"
+  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")
+target_link_libraries(${library_name} PUBLIC
```

Comment:
Question: when do you know when to do `geometry_msgs_TARGETS` vs setting a library like `tf2_ros::tf2_ros`?

Reply:
Right, good question.

In general, you should always prefer to use the target name, like `tf2_ros::tf2_ros`.  That is the most CMake-like thing to do.

In the current Rolling code, the exception is for messages/services/actions/interfaces packages.  There, there are multiple targets that need to be linked against for things to work, so we need to use `${geometry_msgs_TARGETS}`.  We hope to eventually fix that so that there is a "single" target for message packages, but we aren't there yet.